tests/lib/linter/linter.js

```diff
@@ -2957,6 +2957,18 @@ var a = "test2";
             assert.strictEqual(messages.length, 0);
         });
 
+        // https://github.com/eslint/eslint/issues/14652
+        it("should not report a violation", () => {
+            const code = "/*eslint-env es6\n */ new Promise();";
+
+            const config = { rules: { "no-undef": 1 } };
+
+            const messages = linter.verify(code, config, filename);
+
+            assert.strictEqual(messages.length, 0);
+        });
+
```

Comment:
It would be great to also have a test with multiple environment names in the same `/*eslint-env ...*/` comment, but on separate lines.

Reply:
addressed in 0a895ef0515d92546ff5bb9b67b567a0228e8651.